packages/main/src/Popover.js

```diff
@@ -388,21 +393,25 @@ class Popover extends Popup {
 			width = rect.width;
 			height = rect.height;
 
-			return { width, height };
+			return Promise.resolve({ width, height });
 		}
 
 		this.style.visibility = "hidden";
 		this.style.display = "block";
 
-		rect = this.getBoundingClientRect();
+		return new Promise(resolve => {
+			window.requestAnimationFrame(() => {
```

Comment:
This relies on a side effect of something happening and can easily break. What exactly is the work we are waiting for? Could you make it explicitly wait for it?